tests/osaka/eip7883_modexp_gas_increase/conftest.py

```diff
@@ -40,12 +40,38 @@ def call_contract_post_storage() -> Storage:
 
 
 @pytest.fixture
-def call_succeeds() -> bool:
+def total_tx_gas_needed(
+    fork: Fork, modexp_expected: bytes, modexp_input: ModExpInput, precompile_gas: int
+) -> int:
+    """Calculate total tx gas needed for the transaction."""
+    intrinsic_gas_cost_calculator = fork.transaction_intrinsic_cost_calculator()
+    memory_expansion_gas_calculator = fork.memory_expansion_gas_calculator()
+    sstore_gas = fork.gas_costs().G_STORAGE_SET * (len(modexp_expected) // 32)
+    extra_gas = 100_000
+
+    return (
+        extra_gas
+        + intrinsic_gas_cost_calculator(calldata=bytes(modexp_input))
+        + memory_expansion_gas_calculator(new_bytes=len(bytes(modexp_input)))
+        + precompile_gas
+        + sstore_gas
+    )
+
+
+@pytest.fixture
+def exceeds_tx_gas_cap(total_tx_gas_needed: int, fork: Fork) -> bool:
+    """Determine if total gas requirements exceed transaction gas cap."""
+    tx_gas_limit_cap = fork.transaction_gas_limit_cap()
+    return tx_gas_limit_cap is not None and total_tx_gas_needed > tx_gas_limit_cap
```

Comment:
Checking the data type for tx_gas_limit_cap is so annoying XD! Do you think it is a good idea to do the following?

```suggestion
@pytest.fixture
def exceeds_tx_gas_cap(total_tx_gas_needed: int, fork: Fork, env: Environment) -> bool:
    """Determine if total gas requirements exceed transaction gas cap."""
    tx_gas_limit_cap = fork.transaction_gas_limit_cap() or env.gas_limit
    return total_tx_gas_needed > tx_gas_limit_cap
```

Reply:
Added :)